src/javascript/db.js

```diff
@@ -67,51 +98,77 @@ export function getSong(songId) {
       .then((doc) => {
         if (doc.exists) {
           console.log("Song data:", doc.data());
-          resolve(doc.data());
+          resolve(filterSongData(doc));
         } else {
           // doc.data() will be undefined in this case
-          console.log("No such document!");
+          console.error("No such document!");
           reject();
         }
       })
       .catch(function (error) {
-        console.log("Error getting document:", error);
-        reject();
+        console.error("Error getting document:", error);
+        reject(error);
       });
   });
 }
 
+export function updateSong(info) {
+  clean(info);
+  return songsCollection
+    .doc(info.id)
+    .update(info)
+    .then(function () {
+      console.log("Document successfully updated!");
+    })
+    .catch(function (error) {
+      // The document probably doesn't exist.
+      console.error("Error updating document: ", error);
+    });
+}
+
+function clean(obj) {
+  for (let propName in obj) {
+    if (obj[propName] === null || obj[propName] === undefined) {
+      delete obj[propName];
+    }
+  }
+}
+
```

Comment:
This seems... peculiar. Why exactly is this needed?

Reply:
It's the other way around, if the attribute has data, should null overwrite it or not, is null intended by the user or not. I might need to find a better solution T T